src/api/data/cache/cache-helper.go

```diff
@@ -0,0 +1,212 @@
+package cache
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/helm/monocular/src/api/swagger/models"
+)
+
+// Downloads the chart tar file linked by metadata.Urls and store
+// the wanted files (i.e README.md) under chartDataDir
+func downloadAndExtractChartTarball(chart *models.ChartPackage) error {
+	if err := ensureChartDataDir(chart); err != nil {
+		return err
+	}
+
+	if !tarballExists(chart) {
+		if err := downloadTarball(chart); err != nil {
+			return err
+		}
+	}
+
+	if err := extractFilesFromTarball(chart); err != nil {
```

Comment:
tiny nit: you could just simplify this to `return extractFilesFromTarball(chart)`